### ISSUE TITLE
[feature]: Allow disabling registration for self-hosted instances

### ISSUE THREAD

**Issue by frereit** [makeplane/plane#779]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

When self-hosting, I do not want to allow anyone to register and create a workspace. Only invited users should be able to register, or it should be possible to completely disable registration so that once all users are onboarded registration can be disabled.

### Why should this be worked on?

Without this feature, anyone hosting an instance must potentially maintain workspaces for other teams, or allows other people to store data on their server, which is not desirable. Additionally, when processing other users data in the EU, GDPR compliance is required so the hoster would have to deal with potential legal issues.

**Comment by Quadrubo** [comment]
Any update on this issue, currently have to run plane behind http basic auth which is pretty annoying and half broken

**Comment by pablohashescobar** [comment]
Hey @Quadrubo, we are still actively working on this task as it requires significant changes to the current onboarding workflow. Due to the complexity involved, it is taking some time to complete. Rest assured, we will keep the community updated on the progress. Sorry for the inconvenience caused.

**Comment by pablohashescobar** [comment]
Hi there! I'd like to update you on the current version of Plane (0.8). We have introduced a new feature that allows you to toggle signups using the ENABLE_SIGNUP environment variable. This feature provides you with control over enabling or disabling signups based on your requirements.

I'm happy to inform you that the reported issue has already been addressed and resolved in the latest version. Therefore, we are closing this issue. However, if you encounter any difficulties or have any further concerns, please don't hesitate to reopen the issue.

**Comment by Kraego** [comment]
@pablohashescobar maybe I didn't get it right but setting the enable signup variable has no effect on my instance `HOMECHART_APP_ENABLE_SIGNUPD: "false"`. I assume the prefix for the environment variable is needed?

**Comment by tranminhnhanchinh** [comment]
That's issue is solved. But there is a problem that's users who are invited can create new workspaces. How can i lock this?

**Comment by andrey-castro-outburst** [comment]
I have the same issue as @tranminhnhanchinh. @pablohashescobar any updated on this issue? Invited users shouldn't be able to create their own workspaces freely.

**Comment by srinivaspendem** [comment]
Hello, we are yet to prioritise this issue. Can you let me know what are the concerns here which makes you suggest we should block invited users from creating workspaces of their own?

**Comment by andrey-castro-outburst** [comment]
@srinivaspendem How I see it, if I invite someone to my plane instance, it can be just a regular user I want to be able to create/follow tickets on a project, but I don't want everyone I invite in my instance to be able to freely create their own workspaces. That can consume resources on my instance easily, and also it can be a way for the tool to be utilized in a way I didn't want it to be used, like tracking personal project items for users.

**Comment by Nathan13888** [comment]
I am noticing that if you created an instance with the option **enabled originally**. Changing it after initialization **does NOT** disable user registrations.

**Comment by pablohashescobar** [comment]
@Nathan13888, the option only works when the registering user does not have an invite. Also which method were you using to sign up - OAuth, Magic Code, Email Password?

**Comment by Nathan13888** [comment]
@pablohashescobar So it's the `ENABLE_SIGNUP` option. Invited users are expected to be able to sign up an account; however, a user on the internet with access to the login page shouldn't have an account automatically created otherwise :)

I ended up going into Postgres and modifying the `instance_config` (or named smth like that) table and modified that variable, so now it works 🥲

**Comment by pablohashescobar** [comment]
@Nathan13888, you can set this value in the god mode also.

**Comment by Nathan13888** [comment]
Oh hmm, I checked God mode before (and just now). I don't see that option for signups.

**Comment by prateekshourya29** [comment]
@Nathan13888, You can utilize the toggle labeled "Let your users log in via the methods below" in the "SSO and OAuth" tab to deactivate sign-up functionality. Let us know in case you need any further assistance.

**Comment by MichaelErmer** [comment]
Its not working, IMO.
I have disabled the flag you mentioned @prateekshourya29, I also set ENABLE_SIGNUP=0 in the env file and furthermore, I validated the value in the instances_configurations, its `ENABLE_SIGNUP           | 0`, however I can still use a random email at "login" and it gets a code via email which can then be use to login and create a new workspace.

**Comment by stefanux** [comment]
feature completed ( ENABLE_SIGNUP=0 in .env), imho close.

**Comment by MeowSaiGithub** [comment]
is it in the master branch or other branch: i could not find the option in the god mode.

**Comment by almereyda** [comment]
When [searching for `ENABLE_SIGNUP` in this repository](https://github.com/search?q=repo%3Amakeplane%2Fplane%20ENABLE_SIGNUP&type=code) I am pointed towards these resources:

- [plane/ENV_SETUP.md at c24be25024a164ff50362ed7c102a1e101d9e7b9 · makeplane/plane](https://github.com/makeplane/plane/blob/c24be25024a164ff50362ed7c102a1e101d9e7b9/ENV_SETUP.md?plain=1#L118) < This is where the usage of the setting is explained.
- [plane/deploy/selfhost/README.md at c24be25024a164ff50362ed7c102a1e101d9e7b9 · makeplane/plane](https://github.com/makeplane/plane/blob/c24be25024a164ff50362ed7c102a1e101d9e7b9/deploy/selfhost/README.md?plain=1#L305) < This is where we are shown how it looks like when the variable is loaded from the environment.

Do these places help you further to learn how to set the `ENABLE_SIGNUP` variable to `0`? It's a variable that you set in the `apiserver/.env` file.

You are right that it is not included in [the example file](https://github.com/makeplane/plane/blob/preview/apiserver/.env.example). Maybe it could a good way to add it there with the default value `1`, in so it is easier for people to find out how to change it.

The example in `ENV_SETUP.md` could also be synchronised with the `apiserver/.env.example`, in so all available options are explained. For now there is no single list of environmental variables _in this repository_ and also not _in the documentation_ that would explain how to use them.

- [Plane Documentation - Plane](https://docs.plane.so/introduction/home)
- [makeplane/docs: The official https://docs.plane.so documentation 📝](https://github.com/makeplane/docs)

I had also struggled to find my way around configuring the plane instance with environmental variables, due to this lack of documentation. Maybe it could be a good target for this issue to close when this kind of documentation is available?

**Comment by prateekshourya29** [comment]
Hi there, we've added this feature to god-mode. Update to the latest version of plane to access it. Once updated, go to the `Authentication` tab and you'll see a toggle labeled `Allow anyone to sign up without invite`. Simply toggle it off to deactivate sign-up functionality. Check out the attached screenshot for more details.

![image](https://github.com/makeplane/plane/assets/33979846/6327efe8-b985-4d8f-94ed-42d6d58158ad)

**Comment by MeowSaiGithub** [comment]
I know this is closed and i also found that option but unfortunately only on preview release, not stable release as per document - https://docs.plane.so/self-hosting/methods/docker-compose  ,

**Comment by prateekshourya29** [comment]
Yes right @MeowSaiGithub!, it's in the preview release. We'll have it in the stable version soon. Thanks for your understanding!

**Comment by HRNPH** [comment]
Is the ENABLE_SIGNUP updated on the docker stable version?
as I still can't enable the features. will work on it myself if I know where to look.

**Comment by MeowSaiGithub** [comment]
Use preview version from their official documentation. That's work for me currently
